packages/autocomplete-plugin-algolia-insights/src/createSearchInsightsApi.ts

```diff
@@ -24,7 +30,39 @@ function chunk<TItem extends { objectIDs: string[] }>(
   return chunks;
 }
 
+function mapToInsightsParamsApi<
+  TInsightsParamsType extends {
+    items: AlgoliaInsightsHit[];
+    objectIDs?: string[];
+  }
+>(params: TInsightsParamsType[]) {
+  return params.map(({ items, ...param }) => ({
+    ...param,
+    objectIDs: items?.map(({ objectID }) => objectID) || param.objectIDs,
+  }));
+}
+
 export function createSearchInsightsApi(searchInsights: InsightsClient) {
+  const canSendHeaders = isModernInsightsClient(searchInsights);
+
+  function sendToInsights<TInsightsMethod extends InsightsClientMethod>(
+    method: InsightsClientMethod,
+    payloads: InsightsMethodMap[TInsightsMethod],
+    items?: AlgoliaInsightsHit[]
+  ) {
+    if (canSendHeaders && typeof items !== 'undefined') {
+      const { appId, apiKey } = items[0].__autocomplete_algoliaCredentials;
+      const headers = {
+        'X-Algolia-Application-Id': appId,
+        'X-Algolia-API-Key': apiKey,
+      };
+
+      searchInsights(method, ...payloads, { headers } as any);
```

Comment:
I guess it's because we haven't updated the `search-insights` dependency to `2.4.0` yet, as this implies some changes that we want to address in a new PR. Am I right @dhayab?

Reply:
Exactly. Updating the requirements of `search-insights` to 2.4.0 creates a lot of issues with other Insights tests. To prevent adding too much noise I casted the additional parameter as any in the meantime.